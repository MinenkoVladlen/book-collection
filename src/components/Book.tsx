import React from 'react';
import { Avatar, ListItem, ListItemAvatar, ListItemText } from '@mui/material';
import { Book as BookIcon } from '@mui/icons-material';

interface BookProps {
    book: {
        name: string;
        author: string;
    }
}

const Book: React.FC<BookProps> = ({ book }) => {
    return (
        <ListItem>
            <ListItemAvatar >
                <Avatar>
                    <BookIcon />
                </Avatar>
            </ListItemAvatar>
            <ListItemText primary={book.name} secondary={book.author} />
        </ListItem>
    )
}

export default Book;