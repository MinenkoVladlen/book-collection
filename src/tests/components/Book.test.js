import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Book from '../../components/Book';
import booksStore from '../../services/books/Books.store';

jest.mock('../../services/books/Books.store');

describe('Book', () => {
   const book = {
      name: 'Test Book',
      author: 'Test Author',
    }

    it('should render correctly', () => {
        const { asFragment } = render(<Book book={book} />);
        expect(asFragment()).toMatchSnapshot();
    });
});