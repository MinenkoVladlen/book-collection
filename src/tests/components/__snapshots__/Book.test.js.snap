// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Book should render correctly 1`] = `
<DocumentFragment>
  <li
    class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-cfd5wn-MuiListItem-root"
  >
    <div
      class="MuiListItemText-root MuiListItemText-multiline css-14ln1j6-MuiListItemText-root"
    >
      <span
        class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-rizt0-MuiTypography-root"
      >
        Test Book
      </span>
      <p
        class="MuiTypography-root MuiTypography-body2 MuiListItemText-secondary css-1a1whku-MuiTypography-root"
      >
        Test Author
      </p>
    </div>
  </li>
</DocumentFragment>
`;
